Add `android_prune_orphaned_translations` action (#734)
wordpress-mobile/release-toolkit: CHANGELOG.md

### New Features

> - Added `android_prune_orphaned_translations` action: removes `<string>`, `<string-array>` and `<plurals>` entries from `values-*/strings.xml` whose key is not declared in the source strings (the res dir's default `values/strings.xml`, optionally unioned with `additional_source_strings_paths`). Useful after downloading translations from GlotPress to avoid Lint `ExtraTranslation` errors from keys removed/renamed since the source was last synced. [#734]

Reviewer: Formatting nitpick. This is my personal opinion, but I find the long changelog entries AI makes noisy. I find the kind of shorter entries we used to make manually easier to read. Or, rather, I find these AI-generated ones easier to gloss over without taking them in.

```suggestion
- `android_prune_orphaned_translations` action: removes `<string>`, `<string-array>` and `<plurals>` entries from `values-*/strings.xml` whose keys are not declared in the source `values/strings.xml optionally unioned with `additional_source_strings_paths`. [#734]
```
